src/HatsSignerGate.sol

```diff
@@ -407,14 +413,19 @@ contract HatsSignerGate is
     // ensure that the call is coming from the safe
     if (msg.sender != address(safe)) revert NotCalledFromSafe();
```

Comment:
would need to think about this more but want to get it down for later...

i wonder if we still need this check now that we have the more sophisticated reentrancy guard and we're using transient storage